### DISCUSSION TITLE
make the styled-jsx swc plugin can load custom path

### DISCUSSION THREAD

**Discussion by ahabhgk** [vercel/next.js#37077] [Ideas]
### Describe the feature you'd like to request

The styled-jsx swc plugin loads `_JSXStyle` from `next/dist/shared/lib/styled-jsx`, and this path is hard-cded for now, this leads the awesome swc plugin can only used on next.js project. It would be great if this path is configurable.

https://github.com/vercel/next.js/blob/5acf9db617747389c7c2ff7eeafa7492086f01f9/packages/next-swc/crates/styled_jsx/src/utils.rs#L313

### Describe the solution you'd like

A custom path can be passed at here as a parameter.

https://github.com/vercel/next.js/blob/5acf9db617747389c7c2ff7eeafa7492086f01f9/packages/next-swc/crates/styled_jsx/src/lib.rs#L24

I'd like to implements it if this feature request is accepted.

### Describe alternatives you've considered

N/A

**Comment by kwonoj** [comment]
What prevents to use swc's plugin to support out-of-next.js usage? https://github.com/swc-project/plugins/tree/main/packages/styled-jsx
